hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/EntityRecordItemListener.java

```diff
@@ -337,16 +342,26 @@ private com.hederahashgraph.api.proto.java.NftTransfer findNftTransferInsideBody
     }
 
     private void insertFungibleTokenTransfers(
-            long consensusTimestamp, TransactionBody body, boolean isTokenDissociate,
-            TokenID tokenId, EntityId entityTokenId, EntityId payerAccountId, List<AccountAmount> tokenTransfers) {
-        boolean isDeletedTokenDissociate = isTokenDissociate && tokenTransfers.size() == 1;
+            RecordItem recordItem, EntityId tokenId,
+            EntityId payerAccountId, List<AccountAmount> tokenTransfers) {
+        long consensusTimestamp = recordItem.getConsensusTimestamp();
+        TransactionBody body = recordItem.getTransactionBody();
+        boolean isTokenDissociate = body.hasTokenDissociate();
+        int tokenTransferCount = tokenTransfers.size();
+
+        boolean isDeletedTokenDissociate = isTokenDissociate && tokenTransferCount == 1;
+
+        boolean isWipeOrBurn = (recordItem.getTransactionType() == TransactionType.TOKENBURN.getProtoId() || recordItem.getTransactionType() == TransactionType.TOKENWIPE.getProtoId()) && !isDeletedTokenDissociate;
+        boolean isMint = recordItem.getTransactionType() == TransactionType.TOKENMINT.getProtoId() && !isDeletedTokenDissociate;
```

Comment:
This is misleading and unnecessary. If the type is burn or wipe transaction, it can't also be a dissociate transaction. You can just remove all dissociate logic for now since we're checking transfer size is 2 for single transfer.
```suggestion
        boolean isWipeOrBurn = recordItem.getTransactionType() == TransactionType.TOKENBURN.getProtoId() || recordItem.getTransactionType() == TransactionType.TOKENWIPE.getProtoId();
        boolean isMint = recordItem.getTransactionType() == TransactionType.TOKENMINT.getProtoId();
```

Reply:
Done.